packages/adapters-library/src/unwrapCache.ts

```diff
@@ -47,6 +57,34 @@ export class UnwrapCache implements IUnwrapCache {
 
     await this.unwrapCacheProvider.setToDb(key, value)
 
+    return value
+  }
+  async fetchPriceWithCache(
```

Comment:
It might be possible to make an abstraction in which instead of the adapter and the input, we pass a runner function, chainId, token address and block number, but I don't really think it's worth it at this point.

Maybe if we revisit things in the future.